examples/depool-statistics/index.js

```diff
@@ -0,0 +1,171 @@
+    const count = Math.min(3, participantCount);
+    console.log(`\n*** First ${count} Participants of ${participantCount} ***`);
+    for (let i = 0; i < count; i++) {
+        await printParticipant(dePoolAcc, participants[i]);
+    }
+}
+
+function printStakes(heading, stakes) {
+    let entries = Object.entries(stakes);
+    console.log(`  ${heading}:`, entries.length === 0 ? "(none)" : "");
+    for (let [key, value] of entries) {
+        console.log(`    Round #${key}: ${value} nanotokens`);
+    }
+}
+
+function printSpecialStakes(heading, stakes) {
+    let entries = Object.entries(stakes);
+    console.log(`  ${heading}:`, entries.length === 0 ? "(none)" : "");
+    for (let [key, value] of entries) {
+        console.log(`    Round #${key}:`);
+        console.log("      Owner:", value.owner);
+        console.log(`      Remaining Amount: ${value.remainingAmount} nanotokens`);
+        console.log("      Last Withdrawal Time:", unixTimeToString(parseInt(value.lastWithdrawalTime)));
+        console.log(`      Withdrawal Period: ${value.withdrawalPeriod} seconds`);
+        console.log(`      Withdrawal Value: ${value.withdrawalValue} nanotokens`);
+    }
+}
+
+async function printParticipant(dePoolAcc, participant) {
+    const response = await dePoolAcc.runLocal("getParticipantInfo", { addr: participant });
+    const info = response.decoded.output;
+    console.log(`\nParticipant ${participant}:`);
+    console.log(`  Participant Total Stake: ${info.total} nanotokens`);
+    console.log(`  Withdraw Value: ${info.withdrawValue} nanotokens`);
+    console.log("  Reinvest:", info.reinvest);
+    console.log(`  Reward: ${info.reward} nanotokens`);
+    console.log("  Vesting Donor:", info.vestingDonor);
+    console.log("  Lock Donor:", info.lockDonor);
+
+    printStakes("Stakes", info.stakes);
+    printSpecialStakes("Vestings", info.vestings);
+    printSpecialStakes("Locks", info.locks);
+}
+
+function printRound(number, info) {
+    console.log(`\nRound #${number}:`);
+    console.log("  Participant Count:", info.participantQty);
+    console.log(`  Stake: ${info.stake} nanotokens`);
+    console.log(`  Recovered Stake: ${info.recoveredStake} nanotokens`);
+    console.log(`  Unused: ${info.unused} nanotokens`);
+    console.log(`  Validator Stake: ${info.validatorStake} nanotokens`);
+    console.log(`  Validator Remaining Stake: ${info.validatorRemainingStake} nanotokens`);
+}
+
+function unixTimeToString(unixTimestamp) {
+    return new Date(unixTimestamp * 1000).toLocaleString();
+}
+
+async function main(client) {
+    const dePoolCodeHashes = [
+        "b4ad6c42427a12a65d9a0bffb0c2730dd9cdf830a086d94636dab7784e13eb38", // DePool v1
+        "a46c6872712ec49e481a7f3fc1f42469d8bd6ef3fae906aa5b9927e5a3fb3b6b", // DePool v2
+        "14e20e304f53e6da152eb95fffc993dbd28245a775d847eed043f7c78a503885", // DePool v3 Final
+    ];
+
+    // Count all known DePool contracts
+    const dePoolCount = (await client.net.query({
```

Comment:
you can use here: https://github.com/tonlabs/TON-SDK/blob/master/docs/mod_net.md#aggregate_collection